great_expectations/execution_engine/execution_engine.py

```diff
@@ -402,6 +470,16 @@ def add_column_row_condition(
         new_domain_kwargs["row_condition"] = f'col("{column}").notnull()'
         return new_domain_kwargs
 
+    def resolve_data_reference(
+        self, data_connector_name: str, template_arguments: dict
+    ):
+        """Resolve file path for a (data_connector_name, execution_engine_name) combination."""
+        return DataConnectorStorageDataReferenceResolver.resolve_data_reference(
```

Comment:
Rather than have an AttributeError get thrown here (or bubbled up from `resolve_data_reference`) and then wrapping it as a DataConnectorError in each caller, can we:

- Add a `contains_data_reference` to the `DataConnectorStorageDataReferenceResolver` method.
- `if not contains ...` raise `DataConnectorError` with a clear message: "Data Connector <type name> is not supported by Execution Engine <engine name>"

This will further reduce code duplication and provide a clearer error message

Reply:
@donaldheppner The error would not be thrown in the `ExecutionEngine` -- the try/catch is placed into each `DataConnector` subclass that calls `self.execution_engine` in order to detect non-existent or corrupted `ExecutionEngine` instance.  Then the `DataConnectorError` is thrown, because it is expected in such cases.  Thanks.